api4/command.go

```diff
@@ -362,6 +363,39 @@ func listAutocompleteCommands(c *Context, w http.ResponseWriter, r *http.Request
 	w.Write([]byte(model.CommandListToJson(commands)))
 }
 
+func listCommandAutocompleteSuggestions(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+	if !c.App.SessionHasPermissionToTeam(*c.App.Session(), c.Params.TeamId, model.PERMISSION_VIEW_TEAM) {
+		c.SetPermissionError(model.PERMISSION_VIEW_TEAM)
+		return
+	}
+
+	roleId := model.SYSTEM_USER_ROLE_ID
+	if c.IsSystemAdmin() {
+		roleId = model.SYSTEM_ADMIN_ROLE_ID
+	}
+
+	userInput := r.URL.Query().Get("user_input")
+	if len(userInput) == 0 {
+		c.SetInvalidParam("userInput")
+		return
+	}
+	userInput = strings.TrimPrefix(userInput, "/")
+
+	commands, err := c.App.ListAutocompleteCommands(c.Params.TeamId, c.App.T)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	suggestions := c.App.GetSuggestions(commands, userInput, roleId)
+
+	w.Write(model.AutocompleteSuggestionsToJSON(suggestions))
```

Comment:
not sure how it's handled in the server code generally, but shouldn't the `Content-Type` be set here?

Reply:
Content type is set by default for all REST API calls [here](https://github.com/mattermost/mattermost-server/blob/182c29b456eeb3df9a9c757dcc630995d537a44c/web/handlers.go#L158-L159)